### ISSUE TITLE
Ensure that the app gracefully continues if the refresh token does not work

### ISSUE THREAD

**Issue by michaellperry** [jinaga/BlogMaui#15]
The app fails when it cannot refresh the expired token.

```
111|2024-03-01T15:37:28.5945590+00:00|ERROR|7|NetworkManager|Fetch failed after 247 ms. --> System.Exception: Failed to refresh the token: 503 Service Unavailable
   at Jinaga.Maui.Authentication.OAuthClient.RequestNewToken(String refreshToken)
   at Jinaga.Maui.Authentication.OAuth2HttpAuthenticationProvider.RefreshToken()
   at Jinaga.Maui.Authentication.OAuth2HttpAuthenticationProvider.<Reauthenticate>b__11_0()
   at Jinaga.Maui.Authentication.OAuth2HttpAuthenticationProvider.<Lock>d__15`1[[System.Boolean, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at Jinaga.Http.HttpConnection.<WithHttpClient>d__12`1[[Jinaga.Http.FeedsResponse, Jinaga, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
   at Jinaga.Http.HttpNetwork.Feeds(FactReferenceTuple givenTuple, Specification specification, CancellationToken cancellationToken)
   at Jinaga.Managers.NetworkManager.Fetch(FactReferenceTuple givenTuple, Specification specification, CancellationToken cancellationToken)
```

See if this also happens while offline. Ensure that the user can see locally stored data and continue to work until the connection is restored.

**Comment by michaellperry** [comment]
The app remains logged in and functional. The status indicator changes to yellow after attempting to load, and red if the queue is not empty.

The replicator returns a 400 error.

The user might not know that the fix is logging out and logging back in again. Perhaps the app should respond by automatically logging the user out and displaying the login page if it receives a 4xx error while attempting a token refresh. This indicates that the app is able to connect, and that logging in again is a likely fix.

Perhaps it should be sensitive to the expiration date of the refresh token.

**Comment by michaellperry** [comment]
Addressed in PR #25
